src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -104,33 +107,41 @@ public void Start()
             stopping = false;
             pipelineInvocationTasks = new ConcurrentDictionary<Task, Task>();
 
-            internalReceiver = clientEntities.Get(fullPath, entity.Namespace.Alias);
-
-            if (internalReceiver == null)
+            
+            for(var i = 0; i < numberOfClients; i++)
             {
-                throw new Exception($"MessageReceiverNotifier did not get a MessageReceiver instance for entity path {fullPath}, this is probably due to a misconfiguration of the topology");
-            }
 
-            Func<BrokeredMessage, Task> callback = message =>
-            {
-                var processTask = ProcessMessageAsync(message);
-                pipelineInvocationTasks.TryAdd(processTask, processTask);
-                processTask.ContinueWith(t =>
+                var internalReceiver = clientEntities.Get(fullPath, entity.Namespace.Alias);
+
+                if (internalReceiver == null)
                 {
-                    Task toBeRemoved;
-                    pipelineInvocationTasks.TryRemove(t, out toBeRemoved);
-                }, TaskContinuationOptions.ExecuteSynchronously);
-                return processTask;
-            };
+                    throw new Exception($"MessageReceiverNotifier did not get a MessageReceiver instance for entity path {fullPath}, this is probably due to a misconfiguration of the topology");
+                }
 
-            isRunning = true;
+                Func<BrokeredMessage, Task> callback = message =>
+                {
+                    var processTask = ProcessMessageAsync(internalReceiver, message);
+                    pipelineInvocationTasks.TryAdd(processTask, processTask);
+                    processTask.ContinueWith(t =>
+                    {
+                        Task toBeRemoved;
+                        pipelineInvocationTasks.TryRemove(t, out toBeRemoved);
+                    }, TaskContinuationOptions.ExecuteSynchronously);
+                    return processTask;
+                };
+
+                isRunning = true;
```

Comment:
This is not part of this PR, but I wonder if we better off to set `isRunning` to `true` when we're done with the receiver? I.e. after `internalReceiver.OnMessage(callback, options);` is invoked?